db/backup/backup.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3.7
+
+""" Daily back up function for databases within a local
+MariaDB instance """
+
+from datetime import datetime
+import subprocess
+from google.cloud import storage
+from google.cloud import logging
+import pytz
+
+
+def perform_backup():
+    """Completes a backup of the databases within a local mariadb instance
+    and uploads this backup to GCS."""
+
+    # Logging
+    logging_client = logging.Client()
+    log_name = 'backup_log'
+    logger = logging_client.logger(log_name)
+
+    # Get timestamp
+    utc_now = pytz.utc.localize(datetime.utcnow())
+    timestamp_str = utc_now.strftime('%d_%m_%Y_%H-%M-%S')
+
+    # Export SQL Data
+    try:
+        backup_sql = subprocess.check_output(
```

Comment:
Probably want to write this to a temporary file, otherwise we're keeping the whole db in memory, to then write. Looking at the tempfile module in Python, you could create a tmpdir name, wrap it in a `with` (context block) and it get's cleaned up automatically.

```python
with tempfile.TemporaryFile() as fp:
    result = subprocess.run([command], stdout=fp)
    if result.rc != 0:
        # fail state
        pass
    else:
        # success
        blob.upload_from_filename(filepath)
```

Reply:
This suggestion was really useful, but I had issues implementing it with the new mariadb backup function. The created folder is quite difficult to remove without using a sudo rm. Would be very interested if you knew of an alternative fix though.